include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -1008,24 +1041,39 @@ FEPointEvaluation<n_components, dim, spacedim, Number>::evaluate(
 
       const auto unit_points =
         mapping_info->get_unit_points(current_cell_index, current_face_number);
+      const auto &mapping_data =
+        mapping_info->get_mapping_data(current_cell_index, current_face_number);
 
       const std::size_t n_points = unit_points.size();
       const std::size_t n_lanes  = VectorizedArray<Number>::size();
-      for (unsigned int i = 0; i < n_points; i += n_lanes)
+      for (unsigned int i = 0, qb = 0; i < n_points; i += n_lanes, ++qb)
         {
-          // convert to vectorized format
-          Point<dim, VectorizedArray<Number>> vectorized_points;
-          for (unsigned int j = 0; j < n_lanes && i + j < n_points; ++j)
-            for (unsigned int d = 0; d < dim; ++d)
-              vectorized_points[d][j] = unit_points[i + j][d];
-
           // compute
-          const auto val_and_grad =
-            internal::evaluate_tensor_product_value_and_gradient(
-              poly,
-              solution_renumbered,
-              vectorized_points,
-              polynomials_are_hat_functions);
+          const unsigned int n_shapes     = poly.size();
+          const auto         val_and_grad = [&]() {
+            if (polynomials_are_hat_functions)
```

Comment:
I am usually not a big fan of using lambda functions in this way, but I think the ability to use `auto` here is an example of where it can make sense.

Reply:
I also do not really like it, especially without the `std::invoke()`, but I went for this solution here to keep the `const auto`.